cmd/mdatagen/loader.go

```diff
@@ -100,20 +100,22 @@ type templateContext struct {
 	Package string
 }
 
-func loadMetadata(ymlData []byte) (metadata, error) {
-	var out metadata
-
-	// Unmarshal metadata.
-	if err := yaml.Unmarshal(ymlData, &out); err != nil {
-		return metadata{}, fmt.Errorf("unable to unmarshal yaml: %v", err)
+func loadMetadata(filePath string) (metadata, error) {
+	mdMap, err := config.NewMapFromFile(filePath)
```

Comment:
Use the configmapprovider please. See https://github.com/open-telemetry/opentelemetry-collector/pull/4502/

Reply:
done